test/boost/cql_query_test.cc

```diff
@@ -5842,6 +5842,20 @@ SEASTAR_TEST_CASE(test_sending_tablet_info_unprepared_select) {
     }, tablet_cql_test_config());
 }
 
+// Reproduces #20768
```

Comment:
https://github.com/scylladb/scylladb/pull/20208, which has a workaround for this bug in the form of a boolean flag that skips updating init_tablets in `get_old_options_flattened`, has been queued. This workaround can be removed with this fix.
As an extra task: consider adding additional test for ALTER KS - the tests would check if not mentioning `durable_writes` in ALTER leaves this option unmodified. Perhaps there's such a test already?